app/agents/voice/breeze_buddy/workflows/order_confirmation/websocket_bot.py

```
@@ -773,60 +775,47 @@ async def _handle_address_incorrect(self):
-
-        # Mapping of part_type → index in parts list
-        mapping = {"street": 0, "locality": 1, "landmark": 2, "city": 3, "pincode": 4}
-
-        # Update the specified part
-        if part_type in mapping:
-            parts[mapping[part_type]] = extracted_value
-
-        # Remove empty parts and join with comma (no spaces for webhook format)
-        final_parts = [p for p in parts if p.strip()]
-        self.updated_address = ",".join(final_parts)
+    def _update_address_field(self, field_name: str, value, expected_key: str = None):
```

Ruff (0.13.3)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)